PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/KustoQueriesWithTimeIntervalInQueryStringCheck.java

```diff
@@ -0,0 +1,255 @@
+
+
+            // If an anti-pattern is detected, register a problem
+            if (foundAntiPattern) {
+                PsiElement parentElement = element.getParent();
+
+                if (parentElement instanceof PsiLocalVariable) {
+                    PsiLocalVariable variable = (PsiLocalVariable) parentElement;
+                    String variableName = variable.getName();
+                    timeIntervalParameters.add(variableName);
+                }
+            }
+        }
+
+        /**
+         * This method handles the local variable by checking the initializer of the variable
+         * If the initializer is a PsiLiteralExpression, the method checks the expression for the anti-patterns
+         * by matching regex patterns with the expression text
+         *
+         * @param variable - the local variable to check
+         */
+        private void handleLocalVariable(PsiLocalVariable variable) {
+            PsiExpression initializer = variable.getInitializer();
+            if (initializer != null && initializer instanceof PsiLiteralExpression) {
+                checkExpression(initializer, variable);
+            }
+        }
+
+        /**
+         * This method handles the polyadic expression by processing the expression to replace the variables with their values
+         * The method then checks the expression for the anti-patterns by matching regex patterns with the expression text
+         *
+         * @param polyadicExpression - the polyadic expression to check
+         */
+        private void handlePolyadicExpression(PsiPolyadicExpression polyadicExpression) {
+            // Process the polyadic expression to replace the variables with their values
+            PsiExpression initializer = polyadicExpression instanceof PsiPolyadicExpression ? polyadicExpression : null;
+            checkExpression(initializer, polyadicExpression);
+        }
+
+        /**
+         * This method handles the method call by checking the parameters of the method call
+         * If the parameter is a reference to a variable, the method checks the variable name
+         * If the variable name is in the list of time interval parameters, the method checks if the method call is an Azure client method call
+         * If the method call is an Azure client method call, the method registers a problem
+         *
+         * @param methodCall - the method call to check
+         */
+        private void checkParameterNames(PsiMethodCallExpressionImpl methodCall) {
+            // check the parameters of the method call
+            PsiExpressionList argumentList = methodCall.getArgumentList();
+            PsiExpression[] arguments = argumentList.getExpressions();
+
+            // for each argument in the method call, check if the argument is a reference to a variable
+            for (PsiExpression argument : arguments) {
+                if (!(argument instanceof PsiReferenceExpression)) {
+                    continue;
```

Comment:
Instead of using `continue` prefer to invert the checks and process only the cases that we need to handle.